packages/strapi-plugin-i18n/config/functions/bootstrap.js

```diff
@@ -26,23 +25,23 @@ module.exports = async () => {
     await getService('locales').setDefaultLocale(DEFAULT_LOCALE);
   }
 
-  Object.values(strapi.models).forEach(model => {
-    if (getService('content-types').isLocalized(model)) {
-      // TODO: support adding lifecycles programmatically or connecting to a database event handler to avoid conflicts with existing lifecycles fonctions
-
-      _.set(model, 'lifecycles.beforeCreate', async data => {
-        if (!data.locale) {
-          data.locale = await getService('locales').getDefaultLocale();
-        }
-      });
-
-      _.set(model, 'lifecycles.afterCreate', async entry => {
-        await getService('localizations').addLocalizations(entry, { model });
-      });
-
-      _.set(model, 'lifecycles.afterUpdate', async entry => {
-        await getService('localizations').updateNonLocalizedFields(entry, { model });
+  Object.values(strapi.models)
+    .filter(model => getService('content-types').isLocalized(model))
+    .forEach(model => {
+      strapi.db.lifecycles.register({
+        model: model.uid,
+        async beforeCreate(data) {
+          await getService('lcoalizations').assignDefaultLocale(data);
+        },
+        async afterCreate(entry) {
+          await getService('localizations').addLocalizations(entry, { model });
+        },
+        async afterUpdate(entry) {
+          await getService('localizations').updateNonLocalizedFields(entry, { model });
+        },
+        async afterDelete(entry) {
+          await getService('localizations').removeEntryFromLocalizations(entry, { model });
```

Comment:
Just wanted to note that the getService syntax makes the code really easier to read. Nice addition, can't wait to see more changes like this with the v4!